doc/TESTING_YOUR_CHANGES.md

```diff
@@ -0,0 +1,67 @@
+# Testing your changes and making sure they work as expected
+
+So you contributed or did something else that warrants testing. Nice work.
+Our tests will already cover a good amount of things that could go wrong,
+but you are expected to test your non-trivial changes in-game.
+Typo fixes and the like usually do not need to be tested.
+
+## Applying your changes locally
+
+A very simple way is to simply place your Cataclysm executable into your
+git repository.
+It will automagically use this altered path to load resources.
+For more advanced situations, there are commandline parameters such as
+`--datadir`, which you can point at your git respository.
+Check `--help` for more information.
+
+Exiting to the main menu and loading your save again will re-read most
+JSON files, so after making changes, make sure to either relaunch
+Cataclysm or reload your save, whatever is more convenient.
+
+## Using the debug menu
+
+**Tip: If you name your character "Debug" (or anything starting with Debug),
+the game will not automatically disable achievements and display the
+"The debug menu will not help you get better at the game" anymore.**
+
+The debug menu is a powerful and convenient tool. If you debug frequently,
+consider globally binding it to a key of your choice, e.g F12.
+Different circumstances need different testing, so messing around with EOCs
+needs different testing than adding more item qualities for example.
+
+### Items
+
+With *Spawning > Spawn item* you can conjure up any item you want.
+This menu is very self-explanatory. If an item name ends with a yellow
+`(S)`, it means that the item description is variable as there are snippets
+(hence the `(S)`) manipulating them. You can choose which snippet to apply.
+A gray `(V)` means something similar, but it is a variant, which alters
+more than just the description. As of writing, these can not yet be
+selected via the debug menu and are randomized.
+
+### Mapgen
+
+Many of the options in the `Map` part of the debug menu are self-explanatory.
+
+The *Map editor* and *Overmap editor* are of particular interest though, yet
+they may not be as straightforward as the other options.
+
+The *Map editor* can be used to alter small amounts of terrain and also
+apply a different overmap terrain without actually touching its ID.
+
+The *Overmap editor* is mainly useful to spawn buildings and other specials.
+For spawning entire buildings or other things occupying multiple overmap
+tiles, type `s` to spawn a special. Spawning individual pieces of overmap
+terrain is usually not as commonly used and less useful in that sense.
+
+Make sure to only spawn overmap specials in the non-red areas, or
```

Comment:
```suggestion
Make sure to only spawn overmap specials in the non-highlighted areas, or
```
IIRC the already generated areas are marked with the highlight tile in the graphical overmap.